### ISSUE TITLE
Submit records with unverified email account

### ISSUE THREAD

**Issue by kazlauskis** [BiologicalRecordsCentre/iRecord#102]
After user registration I think it would be great not to force the email verification process before submitting any records. 

I see it that the verification would still be necessary and the account would expire in a year or two if not verified, but one should be able to start submitting records before checking its email.

This comes from the iRecord App. I would like it to allow our users to skip this action before actually using the app.

**Comment by johnvanbreda** [comment]
I can see this would be useful, but the records submitted must be withheld from public reports and verification until the user actually confirms their email, otherwise we'll get spam. There is a release_status flag which could serve this purpose. We could also develop a page listing these unreleased records (perhaps for accounts that were never confirmed for a month) that allows us to delete spam and release the rest.

**Comment by JimBacon** [comment]
Is that how we treat unregistered users who submit records through iRecord at the moment?

**Comment by johnvanbreda** [comment]
Good point - no its not. We have had a couple of records of spam via the anonymous route, but no more that I am aware of. I guess formulating a fake biological record is too tricky!

**Comment by JimBacon** [comment]
One disadvantage is that if they register using an email address containing a typo, records that they submit will be lost to them (unless we are prepared to undertake some detective work to reunite them and I don't want to be doing that myself).

**Comment by japonicus** [comment]
The use-case in which this issue originally arose was the BSBI's New Year Plant Hunt app, which is a spin-off from the iRecord App.  The NYPH app would be unusual in that typical users will only ever use for app once, for a three hour period - as such anything that got in the way of streamlined access (such as having to confirm account activation by email on a mobile device) would be a particular nuisance.

For this particular project we could probably live with having a small proportion of mistyped email addresses.

**Comment by JimBacon** [comment]
Are they going to be good quality records which you want to present to the verification community? Your description makes me question whether iRecord is the correct destination for them.

**Comment by kazlauskis** [comment]
> One disadvantage is that if they register using an email address containing a typo, records that they submit will be lost to them

I guess that disabling, rather than deleting the account + freeing up its records, would suffice here.

**Comment by JimBacon** [comment]
You're suggesting a system with a built in administrative headache. If users are not going to care about reviewing their records or building up a record history and are just having some fun for a few hours why not do the whole thing without any registration at all?

**Comment by johnvanbreda** [comment]
In response to Jim's question about record quality, this is similar to the Garden Bioblitz where we get lots of first time recorders. Our solution was to mark records with release_status=U (unreleased) which causes them to be withheld from verification and public reports. We then have a report that lists all the unreleased records with buttons for rejecting obviously bad records that can't be salvaged, requesting more info or releasing the good ones so they go to the verification system. This can be used by scheme organisers and volunteers who have some knowledge (enough to sanity check the records) but not enough to act as full verifiers. It saves verifiers being inundated with obvious mistakes.

**Comment by johnvanbreda** [comment]
Another answer for Jim though I'm sure @japonicus can add his thoughts on why bother with registration at all:

- Because active recorders as well as one-off recorders will take part, so the active recorders will want to be able to keep their records list.
- As an engagement exercise, it would be a shame for recorders who are first timers to not be able to continue with iRecord and build a list of records after the Plant Hunt event, should they be smitten by the bug.
- Because although some mistakes will be made in registration, most will give an email address that could be subsequently useful during the verification process.

**Comment by japonicus** [comment]
> Are they going to be good quality records which you want to present to the verification community? Your description makes me question whether iRecord is the correct destination for them.
The participants will be quite a mixture of experienced botanists (probably the majority) along with a substantial number of beginners. As the records are coming from a mobile app (and with the condition that the plant is in flower) I'd expect there to be a lot of photographs of common species and would broadly expect the accuracy  to be reasonable (most of the dodgy records will also be quickly validated by the BSBI as part of the NYPH process).

> If users are not going to care about reviewing their records or building up a record history and are just having some fun for a few hours why not do the whole thing without any registration at all?
From past experience it would be useful to be able to contact people (to provide feedback and to query some odd identifications). It's also a good opportunity to get more BSBI members engaged with iRecord. If a few percent of email addresses are mistyped then given the relatively small scale of this, that is something that can be coped with (or ignored without huge repercussions).

This is going a bit off-topic and I think Karolis' bug is intended to have wider scope than just the NYPH app - I just wanted to provide a specific example of why this was useful.

**Comment by JimBacon** [comment]
If the app had an additional option built in to "Resend Registration Email" combined with the ability to update their email address before hand that might overcome my concern (and possibly improve the registration experience all round).

**Comment by BirenRathod** [comment]
May be useful to read http://indicia-docs.readthedocs.io/en/latest/developing/web-services/user-identifiers.html?highlight=open%20id

**Comment by kazlauskis** [comment]
> If the app had an additional option built in to "Resend Registration Email" combined with the ability to update their email address before hand that might overcome my concern (and possibly improve the registration experience all round).

Yes, this would be great. I think the iRecord App should allow such basic user account updates, though I doubt this would be implemented in any other app.

**Comment by BirenRathod** [comment]
Bare in mind to differentiate then spam, anonymous and genuine users by the time user send "Resend registration email" request, if somebody decide to clean the warehouse user records. To differentiate the genuine user will not a problem for app but warehouse might be ended up with many spam users.
Also, if people find out the way to abuse the app by registering a spam user via app. 

Unless app will register with person mobile number and individual verification code.
